src/intern_blks/ya_intern.c

```diff
@@ -129,6 +131,71 @@ __attribute__ ((gnu_inline)) inline void ya_setup_prefix_suffix(ya_block_t *blk,
 	}
 }
 
+#include <playerctl/playerctl.h>
+#define CHECK_PLAYER_ERROR(PREFIX) \
+	if (e_##PREFIX) { \
+		ya_block_error(blk, "Player error!"); \
+	}
+
+#define PRINT_PAUSED_MSG() \
+	sprintf(startstr,"%s",blk->internal->option[2]);
+
+void ya_int_song(ya_block_t *blk) {
+	char *startstr = blk->buf;
+	size_t prflen = 0;
+	size_t suflen = 0;
+	ya_setup_prefix_suffix(blk, &prflen, &suflen, &startstr);
+
+	GError *e_tmp;
+	GError *e_title;
+	GError *e_artist;
+	PlayerctlPlayer *player;
+	gchar *s;
+
+	if (blk->internal->option[0]==NULL)
+		ya_block_error(blk, "Please set option0 for playerctl (e.g. spotify)!");
+	if (blk->internal->option[1]==NULL)
+		blk->internal->option[1] = " - ";
+	if (blk->internal->option[2]==NULL)
+		blk->internal->option[2] = "Paused";
+
+	while (1) {
+		player = playerctl_player_new(
```

Comment:
Should this be executed outside of the while loop? I'm not too familiar with the playerctl API, but it seems excessive to create a new PlayerctlPlayer on every iteration.

Reply:
the problem is that it seems to be unable to retrieve the information properly without recreating the player every time.
I decided to add a `free(player);` at the end of the loop to avoid too heavy memory usage, however I'll try to profile this in the next time to ensure that there is no better approach and this isn't too slow.

Reply:
ok it seems as this struct needs to be recreated during each iteration.
Does this `free()` approach make any sense or do you know sth better?

Reply:
The [`playerctl-cli` source code](https://github.com/acrisci/playerctl/blob/master/playerctl/playerctl-cli.c#L409) uses `g_object_unref(player)` at the end of its loop.